### PR TITLE
[FLINK-12942][docs-zh] Translate Elasticsearch Connector page into Ch…

### PR DESCRIPTION
**Pull request by XuQianJin-Stars**
## What is the purpose of the change

*(This pull request to Translate "Elasticsearch Connector" page into Chinese.)*

## Brief change log

  - *doc locates in flink/docs/dev/connectors/elasticsearch.zh.md*

## Verifying this change

*(Please pick either of the following options)*

no change to test.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / no)
  - The serializers: (yes / no / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / no / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / no / don't know)
  - The S3 file system connector: (yes / no / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: docs/dev/connectors/elasticsearch.zh.md [hunk 1]
```diff
@@ -334,22 +306,14 @@ env.enableCheckpointing(5000) // checkpoint every 5000 msecs
...
 
-### Handling Failing Elasticsearch Requests
+### 处理失败的 Elasticsearch 请求
 
-Elasticsearch action requests may fail due to a variety of reasons, including
-temporarily saturated node queue capacity or malformed documents to be indexed.
-The Flink Elasticsearch Sink allows the user to specify how request
-failures are handled, by simply implementing an `ActionRequestFailureHandler` and
-providing it to the constructor.
+由于各种原因，Elasticsearch 操作请求可能会失败，包括暂时饱和的节点队列容量或要编入索引的格式错误的文档。Flink Elasticsearch Sink 允许用户指定请求的方式通过简单地实现 `ActionRequestFailureHandler` 来处理失败并将它提供给构造函数。
```
Comment (wuchong) [inline docs/dev/connectors/elasticsearch.zh.md]:
```suggestion
由于各种原因，发给 Elasticsearch 的操作请求可能会失败，如暂时饱和的节点队列容量或索引异常文档。Flink Elasticsearch Sink 允许用户指定如何处理请求失败，通过简单地实现`ActionRequestFailureHandler`接口并传给 ElasticsearchSink 的构造函数即可。
```

File: docs/dev/connectors/elasticsearch.zh.md [hunk 2]
```diff
@@ -410,21 +374,11 @@ input.addSink(new ElasticsearchSink(
...
+请注意，`onFailure` 被调用用于仅在之后仍然发生的故障 `BulkProcessor` 内部完成所有退避重试尝试。默认情况下，`BulkProcessor` 重试最多8次尝试指数退避。有关行为的更多信息内部 `BulkProcessor` 以及如何配置它，请参阅以下部分。
 
-By default, if a failure handler is not provided, the sink uses a
-`NoOpFailureHandler` that simply fails for all kinds of exceptions. The
-connector also provides a `RetryRejectedExecutionFailureHandler` implementation
-that always re-add requests that have failed due to queue capacity saturation.
+默认情况下，如果未提供失败处理程序，则接收器使用a `NoOpFailureHandler` 只是因各种异常而失败。该连接器还提供了 `RetryRejectedExecutionFailureHandler` 实现总是重新添加由于队列容量饱和而失败的请求。
 
 <p style="border-radius: 5px; padding: 5px" class="bg-danger">
 <b>IMPORTANT</b>: Re-adding requests back to the internal <b>BulkProcessor</b>
```
Comment (wuchong) [inline docs/dev/connectors/elasticsearch.zh.md]: This sentence is not translated.

**Comment by XuQianJin-Stars** [comment]
hi @wuchong Thank you very much for being in review so late. Comments addressed.

**Comment by klion26** [comment]
Seems there are some conflicts now, could you please rebase the master to resolve the conflict?

**Comment by XuQianJin-Stars** [comment]
> @XuQianJin-Stars thanks for the contribution. Seems there are some conflicts now, could you please rebase the master to resolve the conflict?

well, let me rebase it.

File: docs/dev/connectors/elasticsearch.zh.md [hunk 3]
```diff
@@ -58,23 +56,19 @@ of the Elasticsearch installation:
...
 </table>
 
-Note that the streaming connectors are currently not part of the binary
-distribution. See [here]({{site.baseurl}}/dev/project-configuration.html) for information
-about how to package the program with the libraries for cluster execution.
+请注意，当前 Flink 二进制安装文件不包括 Elasticsearch 流连接器。有关信息，请参见[此处]({{site.baseurl}}/zh/dev/project-configuration.html)
+关于如何将程序与库打包以供集群执行。
 
-## Installing Elasticsearch
+## 安装 Elasticsearch
```
Comment (klion26) [inline docs/dev/connectors/elasticsearch.zh.md:62]: 对于翻译的标题需要在之前添加一个 `<a>` 标签[1]，这个这样其他引用该标题链接的地方可以不修改
这里可以参考英文版 url 生成的链接来添加，比如这里使用 `<a name="installing-elasticsearch"></a>

[1] https://cwiki.apache.org/confluence/display/FLINK/Flink+Translation+Specifications

File: docs/dev/connectors/elasticsearch.zh.md [hunk 4]
```diff
@@ -58,23 +56,19 @@ of the Elasticsearch installation:
...
 
-Instructions for setting up an Elasticsearch cluster can be found
-[here](https://www.elastic.co/guide/en/elasticsearch/reference/current/setup.html).
-Make sure to set and remember a cluster name. This must be set when
-creating an `ElasticsearchSink` for requesting document actions against your cluster.
+[这里](https://www.elastic.co/guide/en/elasticsearch/reference/current/setup.html)可以找到有关设置Elasticsearch集群的说明。确保设置并记住集群名称。集群名称将在创建一个`ElasticsearchSink`时使用，用于请求对你安装的集群的文档操作。
```
Comment (klion26) [inline docs/dev/connectors/elasticsearch.zh.md:64]:
```suggestion
[这里](https://www.elastic.co/guide/en/elasticsearch/reference/current/setup.html)可以找到有关设置 Elasticsearch集群的说明。确保设置并记住集群名称。集群名称将在创建一个 `ElasticsearchSink` 时使用，用于请求对你安装集群的文档操作。
```

File: docs/dev/connectors/elasticsearch.zh.md [hunk 5]
```diff
@@ -116,7 +110,7 @@ input.addSink(new ElasticsearchSink<>(config, transportAddresses, new Elasticsea
                 .type("my-type")
                 .source(json);
     }
```
Comment (klion26) [inline docs/dev/connectors/elasticsearch.zh.md:112]: 可以的话，建议把注释也翻译一下，这样更容易理解

File: docs/dev/connectors/elasticsearch.zh.md [hunk 6]
```diff
@@ -116,7 +110,7 @@ input.addSink(new ElasticsearchSink<>(config, transportAddresses, new Elasticsea
...
-
+
```
Comment (klion26) [inline docs/dev/connectors/elasticsearch.zh.md:113]: 这个修改格式更好看了 👍

File: docs/dev/connectors/elasticsearch.zh.md [hunk 7]
```diff
@@ -283,43 +277,21 @@ input.addSink(esSinkBuilder.build)
...
-[here](https://www.elastic.co/guide/en/elasticsearch/reference/current/index.html).
-Especially important is the `cluster.name` parameter that must correspond to
-the name of your cluster.
+对于 Elasticsearch（即版本等于或低于5.x）的集群仍使用现已弃用的 `TransportClient` 进行通信，请注意 `ElasticsearchSink` 使用一个由 `String` 构成的 `Map` 来进行参数配置 。这个配置 map 将直接在内部创建使用`TransportClient` 时转发。配置关键参数在 Elasticsearch 文档中[此处](https://www.elastic.co/guide/en/elasticsearch/reference/current/index.html)可以查看。特别重要的是 `cluster.name` 参数必须对应你集群的名称。
 
-For Elasticsearch 6.x and above, internally, the `RestHighLevelClient` is used for cluster communication.
-By default, the connector uses the default configurations for the REST client. To have custom
-configuration for the REST client, users can provide a `RestClientFactory` implementation when
-setting up the `ElasticsearchClient.Builder` that builds the sink.
+对于 Elasticsearch 6.x及更高版本，在内部，使用 `RestHighLevelClient` 进行集群通信。默认情况下，连接器使用 REST 客户端的默认配置。要想自定义 REST 客户端的配置，用户可以在提供 `RestClientFactory` 实现时设置构建 sink 的 `ElasticsearchClient.Builder` 。
```
Comment (klion26) [inline docs/dev/connectors/elasticsearch.zh.md:282]:
```suggestion
对于 Elasticsearch 6.x 及更高版本，在内部，使用 `RestHighLevelClient` 进行集群通信。默认情况下，连接器使用 REST 客户端的默认配置。要想自定义 REST 客户端的配置，用户可以在提供 `RestClientFactory` 实现时设置构建 sink 的 `ElasticsearchClient.Builder` 。
```

File: docs/dev/connectors/elasticsearch.zh.md [hunk 8]
```diff
@@ -283,43 +277,21 @@ input.addSink(esSinkBuilder.build)
...
 
-Internally, each parallel instance of the Flink Elasticsearch Sink uses
-a `BulkProcessor` to send action requests to the cluster.
-This will buffer elements before sending them in bulk to the cluster. The `BulkProcessor`
-executes bulk requests one at a time, i.e. there will be no two concurrent
-flushes of the buffered actions in progress.
+在内部，Flink Elasticsearch Sink 的每个并行实例都使用一个 `BulkProcessor` ，用于向集群发送操作请求。这将在批量发送到集群之前缓冲元素。 `BulkProcessor` 一次执行一个批量请求，即不会有两个并发刷新正在进行的缓冲操作。
 
-### Elasticsearch Sinks and Fault Tolerance
+### Elasticsearch Sink 与容错处理
```
Comment (klion26) [inline docs/dev/connectors/elasticsearch.zh.md:288]: 这里也需要添加 `<a>` 标签

File: docs/dev/connectors/elasticsearch.zh.md [hunk 9]
```diff
@@ -283,43 +277,21 @@ input.addSink(esSinkBuilder.build)
...
-With Flink’s checkpointing enabled, the Flink Elasticsearch Sink guarantees
-at-least-once delivery of action requests to Elasticsearch clusters. It does
-so by waiting for all pending action requests in the `BulkProcessor` at the
-time of checkpoints. This effectively assures that all requests before the
-checkpoint was triggered have been successfully acknowledged by Elasticsearch, before
-proceeding to process more records sent to the sink.
+在启用 Flink 的 Checkpoint 后，Flink Elasticsearch Sink 可以保证至少一次向 Elasticsearch 集群传递操作请求。确实如此所以通过等待 `BulkProcessor` 中的所有待处理操作请求 checkpoints 的时间。这有效地保证了之前的所有请求在触发 checkpoint 之前已被 Elasticsearch 成功接收确认继续处理发送到接收器的更多记录。
 
-More details on checkpoints and fault tolerance are in the [fault tolerance docs]({{site.baseurl}}/zh/learn-flink/fault_tolerance.html).
+更多有关 checkpoint 和容错的详细信息，请参见[容错相关文档]({{site.baseurl}}/zh/learn-flink/fault_tolerance.html)。
```
Comment (klion26) [inline docs/dev/connectors/elasticsearch.zh.md:292]: 这里的链接需要使用 `{%link %}` 的形式来改写一下

File: docs/dev/connectors/elasticsearch.zh.md [hunk 10]
```diff
@@ -336,22 +308,14 @@ env.enableCheckpointing(5000) // checkpoint every 5000 msecs
...
-<b>NOTE</b>: Users can disable flushing if they wish to do so, by calling
-<b>disableFlushOnCheckpoint()</b> on the created <b>ElasticsearchSink</b>. Be aware
-that this essentially means the sink will not provide any strong
-delivery guarantees anymore, even with checkpoint for the topology enabled.
-</p>
+<p style =“border-radius：5px; padding：5px”class =“bg-danger”>
+<b>注意</b>：如果用户希望在创建的 <b>ElasticsearchSink</b> 上禁用 flush，可以通过调用 <b>disableFlushOnCheckpoint()</b> 。注意这基本上意味着即使拓扑的 checkpoint 开启了，sink 也无法提供任何强语义的消息投递保证。</p>
 
-### Handling Failing Elasticsearch Requests
+### 处理失败的 Elasticsearch 请求
```
Comment (klion26) [inline docs/dev/connectors/elasticsearch.zh.md:314]: 这里需要增加 `<a>` 标签

File: docs/dev/connectors/elasticsearch.zh.md [hunk 11]
```diff
@@ -336,22 +308,14 @@ env.enableCheckpointing(5000) // checkpoint every 5000 msecs
...
 
-Elasticsearch action requests may fail due to a variety of reasons, including
-temporarily saturated node queue capacity or malformed documents to be indexed.
-The Flink Elasticsearch Sink allows the user to specify how request
-failures are handled, by simply implementing an `ActionRequestFailureHandler` and
-providing it to the constructor.
+由于各种原因，发给 Elasticsearch 的操作请求可能会失败，如暂时饱和的节点队列容量或索引异常文档。Flink Elasticsearch Sink 允许用户指定如何处理请求失败，通过简单地实现`ActionRequestFailureHandler`接口并传给 ElasticsearchSink 的构造函数即可。
```
Comment (klion26) [inline docs/dev/connectors/elasticsearch.zh.md:316]:
```suggestion
由于各种原因，发给 Elasticsearch 的操作请求可能会失败，如暂时饱和的节点队列容量或索引异常文档。Flink Elasticsearch Sink 允许用户指定如何处理请求失败，通过简单地实现 `ActionRequestFailureHandler` 接口并传给 ElasticsearchSink 的构造函数即可。
```

File: docs/dev/connectors/elasticsearch.zh.md [hunk 12]
```diff
@@ -412,61 +376,36 @@ input.addSink(new ElasticsearchSink(
...
+上面的示例将让 sink 重新添加由于失败而导致的请求队列容量饱和和丢弃请求与格式错误的文档，没有失败了。对于所有其他故障， sink 将失败。如果是 `ActionRequestFailureHandler` 没有提供给构造函数，接收器将因任何类型的错误而失败。
+
+请注意，`onFailure` 被调用用于仅在之后仍然发生的故障 `BulkProcessor` 内部完成所有退避重试尝试。默认情况下，`BulkProcessor` 重试最多8次尝试指数退避。有关行为的更多信息内部 `BulkProcessor` 以及如何配置它，请参阅以下部分。
+
+默认情况下，如果未提供失败处理程序，则接收器使用a `NoOpFailureHandler` 只是因各种异常而失败。该连接器还提供了 `RetryRejectedExecutionFailureHandler` 实现总是重新添加由于队列容量饱和而失败的请求。
+
+<b>重要</b>：将请求重新添加回内部 <b>BulkProcessor</b> 在失败时会导致 checkpoints 变长，sink 也一样需要等待重新添加的请求在 checkpoint 时被刷新。例如，当使用 <b>RetryRejectedExecutionFailureHandler</b> 时，checkpoints 需要等待直到 Elasticsearch 节点队列有足够的容量处理所有等待的请求。这也意味着如果从不重新添加请求成功，checkpoint 将永远不会结束。</p>
+
+
+### Bulk Processor 内部配置
```
Comment (klion26) [inline docs/dev/connectors/elasticsearch.zh.md:388]: 这里需要增加 `<a>` 标签

File: docs/dev/connectors/elasticsearch.zh.md [hunk 13]
```diff
@@ -336,22 +308,14 @@ env.enableCheckpointing(5000) // checkpoint every 5000 msecs
 </div>
 </div>
 
-<p style="border-radius: 5px; padding: 5px" class="bg-danger">
-<b>NOTE</b>: Users can disable flushing if they wish to do so, by calling
-<b>disableFlushOnCheckpoint()</b> on the created <b>ElasticsearchSink</b>. Be aware
-that this essentially means the sink will not provide any strong
-delivery guarantees anymore, even with checkpoint for the topology enabled.
-</p>
+<p style =“border-radius：5px; padding：5px”class =“bg-danger”>
```
Comment (klion26) [inline docs/dev/connectors/elasticsearch.zh.md:311]: 这个地方的渲染有问题，可能是中文符号导致的。
这个有个小技巧：在修改完成后，本地执行 `./docs/docker/run.sh` 然后执行 `./build_docs.sh -p` 打开 localhost:4000  可以看到渲染后的结果

File: docs/dev/connectors/elasticsearch.zh.md [hunk 14]
```diff
@@ -412,61 +376,36 @@ input.addSink(new ElasticsearchSink(
...
-
-Alternatively, you can put the connector's jar file into Flink's `lib/` folder to make it available
-system-wide, i.e. for all job being run.
+上面的示例将让 sink 重新添加由于失败而导致的请求队列容量饱和和丢弃请求与格式错误的文档，没有失败了。对于所有其他故障， sink 将失败。如果是 `ActionRequestFailureHandler` 没有提供给构造函数，接收器将因任何类型的错误而失败。
+
+请注意，`onFailure` 被调用用于仅在之后仍然发生的故障 `BulkProcessor` 内部完成所有退避重试尝试。默认情况下，`BulkProcessor` 重试最多8次尝试指数退避。有关行为的更多信息内部 `BulkProcessor` 以及如何配置它，请参阅以下部分。
+
+默认情况下，如果未提供失败处理程序，则接收器使用a `NoOpFailureHandler` 只是因各种异常而失败。该连接器还提供了 `RetryRejectedExecutionFailureHandler` 实现总是重新添加由于队列容量饱和而失败的请求。
+
+<b>重要</b>：将请求重新添加回内部 <b>BulkProcessor</b> 在失败时会导致 checkpoints 变长，sink 也一样需要等待重新添加的请求在 checkpoint 时被刷新。例如，当使用 <b>RetryRejectedExecutionFailureHandler</b> 时，checkpoints 需要等待直到 Elasticsearch 节点队列有足够的容量处理所有等待的请求。这也意味着如果从不重新添加请求成功，checkpoint 将永远不会结束。</p>
```
Comment (klion26) [inline docs/dev/connectors/elasticsearch.zh.md:385]: 这里是不是漏掉了 `<p>` 标签

File: docs/dev/connectors/elasticsearch.zh.md [hunk 15]
```diff
@@ -412,61 +376,36 @@ input.addSink(new ElasticsearchSink(
...
-
-For the execution of your Flink program, it is recommended to build a
-so-called uber-jar (executable jar) containing all your dependencies
-(see [here]({{site.baseurl}}/dev/project-configuration.html) for further information).
-
-Alternatively, you can put the connector's jar file into Flink's `lib/` folder to make it available
-system-wide, i.e. for all job being run.
+上面的示例将让 sink 重新添加由于失败而导致的请求队列容量饱和和丢弃请求与格式错误的文档，没有失败了。对于所有其他故障， sink 将失败。如果是 `ActionRequestFailureHandler` 没有提供给构造函数，接收器将因任何类型的错误而失败。
+
+请注意，`onFailure` 被调用用于仅在之后仍然发生的故障 `BulkProcessor` 内部完成所有退避重试尝试。默认情况下，`BulkProcessor` 重试最多8次尝试指数退避。有关行为的更多信息内部 `BulkProcessor` 以及如何配置它，请参阅以下部分。
```
Comment (klion26) [inline docs/dev/connectors/elasticsearch.zh.md:381]: 这里改成 "`onFailure` 仅在 `BulkProcessor` 内部完成所有退避重试尝试后仍有问题的情况下被调用“ 会好一些吗？

File: docs/dev/connectors/elasticsearch.zh.md [hunk 16]
```diff
@@ -412,61 +376,36 @@ input.addSink(new ElasticsearchSink(
...
+ * **bulk.flush.backoff.enable**: 是否启用带回退延迟的刷新重试。如果一个或多个操作由于临时的 `EsRejectedExecutionException` 而失败。
+ * **bulk.flush.backoff.type**: 回退延迟的类型，`CONSTANT` 或 `EXPONENTIAL`。
+ * **bulk.flush.backoff.delay**: 回退的延迟量。对于常数回退，这个只是每次重试之间的延迟。对于指数回退，这是初始延迟。
+ * **bulk.flush.backoff.retries**: 重试回退的次数。
+
+更多有关 Elasticsearch 的信息可以在[这里](https://elastic.co)找到。
+
+## 将 Elasticsearch Connector 打包到 Uber-Jar 中
+
+为了执行 Flink 程序，建议构建一个包含所有依赖项的所谓 uber-jar（可执行 jar ）有关详细信息，请参阅[此处]({{site.baseurl}}/zh/dev/project-configuration.html)。
```
Comment (klion26) [inline docs/dev/connectors/elasticsearch.zh.md:407]: 这个链接需要修改成 `{%link %}` 的形式

File: docs/dev/connectors/elasticsearch.zh.md [hunk 17]
```diff
@@ -412,61 +376,36 @@ input.addSink(new ElasticsearchSink(
...
+
+或者，您可以将连接器的 jar 文件放入 Flink 的`lib /`文件夹中以使其可用于所有系统，即所有正在运行的 job 。
```
Comment (klion26) [inline docs/dev/connectors/elasticsearch.zh.md:409]:
```suggestion
或者，您可以将连接器的 jar 文件放入 Flink 的 `lib/` 文件夹中以使其可用于所有系统，即所有正在运行的 job 。
```

**Comment by MartijnVisser** [comment]
Connector has been moved to https://github.com/apache/flink-connector-elasticsearch

**Pull request closed without merging**